src/main/java/frc/team2412/robot/commands/autonomous/AutonomousCommand.java

```diff
@@ -1,6 +1,76 @@
+import frc.team2412.robot.util.GeoConvertor;
+import org.frcteam2910.common.math.RigidTransform2;
+import org.frcteam2910.common.math.Vector2;
+
+import java.util.List;
 
 public class AutonomousCommand extends SequentialCommandGroup {
+    DrivebaseSubsystem drivebaseSubsystem;
+
+    public AutonomousCommand(DrivebaseSubsystem d){
+        drivebaseSubsystem = d;
+    }
+    public Command getAutonomousCommand() {
+        // Create config for trajectory
+        TrajectoryConfig config =
+                new TrajectoryConfig(
+                        Constants.AutoConstants.maxSpeedMetersPerSecond,
+                        Constants.AutoConstants.maxAccelerationMetersPerSecondSquared)
+                        // Add kinematics to ensure max speed is actually obeyed
+                        .setKinematics(Constants.DriveConstants.driveKinematics);
+
+        // An example trajectory to follow.  All units in meters.
+        Trajectory exampleTrajectory =
+                TrajectoryGenerator.generateTrajectory(
+                        // Start at the origin facing the +X direction
+                        new Pose2d(0, 0, new Rotation2d(0)),
+                        // Pass through these two interior waypoints, making an 's' curve path
+                        List.of(new Translation2d(1, 1), new Translation2d(2, -1)),
+                        // End 3 meters straight ahead of where we started, facing forward
+                        new Pose2d(3, 0, new Rotation2d(0)),
+                        config);
+
+        var thetaController =
+                new ProfiledPIDController(
+                        Constants.AutoConstants.PThetaController, 0, 0, Constants.AutoConstants.kThetaControllerConstraints);
+        thetaController.enableContinuousInput(-Math.PI, Math.PI);
+
+        SwerveControllerCommand swerveControllerCommand =
+                new SwerveControllerCommand(
+                        exampleTrajectory,
+                        drivebaseSubsystem::getPoseAsPose, // Functional interface to feed supplier
+                        Constants.DriveConstants.driveKinematics,
+
+                        // Position controllers
+                        new PIDController(Constants.AutoConstants.PXController, 0, 0),
+                        new PIDController(Constants.AutoConstants.PYController, 0, 0),
+                        thetaController,
+                        drivebaseSubsystem::updateModules,
+                        drivebaseSubsystem);
+
+        // Reset odometry to the starting pose of the trajectory.
+        drivebaseSubsystem.resetPose(exampleTrajectory.getInitialPose());
+
+        // Run path following command, then stop at the end.
+        //have to fix this later, the parameters are just placeholders to get program to build
+        return swerveControllerCommand.andThen(() -> drivebaseSubsystem.drive(GeoConvertor.translation2dToVector2(exampleTrajectory.getInitialPose().getTranslation()),
+                0, false));
```

Comment:
We may want to update the parameters to make the robot stop (like the original one which had a translation of zero).